pkg/retrier/retrier.go

```diff
@@ -63,9 +63,14 @@ func WithRetryPolicy(policy RetryPolicy) RetrierOpt {
 // Retry runs the fn function until it either successful completes (not error),
 // the set timeout reached or the retry policy aborts the execution
 func (r *Retrier) Retry(fn func() error) error {
+	if r == nil {
+		return fn()
```

Comment:
Doing so would break certain unit tests that call Retry with a null receiver.  In the previous code, these would successfully enter the loop, call fn() and (assuming it was successful) return the error object it returned, all without attempting to dereference the *Retrier receiver (and panicing as a result of this).

When I added code to log some interesting facts about the retrier upon function entry I dereferenced of the receiver before calling fn(), resulting in the panic.

The lines here formally codify the old behavior: when called w/o a receiver it simply calls fn() once and returns its error code before any derefs of the receiver.

Debatable whether that's the "best" solution.  Alternatively I could "fix" the offending unit tests and make the change you suggested.  I chose this approach so I'd touch the fewest other files in the project.

Reply:
Maybe add a comment to make this clear in the code

Reply:
Done.